client/packages/common/src/ui/forms/JsonForms/components/DateTime.tsx

```diff
@@ -45,11 +49,41 @@ export const datetimeTester = rankWith(5, isDateTimeControl);
 
 const UIComponent = (props: ControlProps) => {
   const [error, setError] = React.useState('');
-  const { data, handleChange, label, path } = props;
+  const { data, handleChange, label, path, uischema } = props;
+  const dateFormatter = useFormatDateTime().customDate;
+
   if (!props.visible) {
     return null;
   }
 
+  const dateOnly = uischema.options?.['dateOnly'] ?? false;
+
+  const inputFormat = !dateOnly ? 'dd/MM/yyyy hh:mm' : 'dd/MM/yyyy';
+
+  const onChange = (e: Date | null) => {
+    if (!e) return;
+
+    try {
+      const dateString = !dateOnly
+        ? e.toISOString()
+        : // By default, will use current date-time. However, if a
+          // different date is selected, the time will be considered "midnight"
+          dateFormatter(e, 'yyyy-MM-dd') + ' 00:00:00';
+      setError('');
+      if (e) handleChange(path, dateString);
+    } catch (err) {
+      setError((err as Error).message);
+      console.error(err);
```

Comment:
Think we can remove this console.error because its a common user error to enter an invalid data and maybe better to not write it to the console (?)

Reply:
Sure.